cli/azd/pkg/tools/tool.go

```diff
@@ -60,11 +61,36 @@ func ExecuteCommand(ctx context.Context, cmd string, args ...string) (string, er
 	return runResult.Stdout, err
 }
 
-func ExtractSemver(cliOutput string) (semver.Version, error) {
-	ver := regexp.MustCompile(`\d+\.\d+\.\d+`).FindString(cliOutput)
-	semver, err := semver.Parse(ver)
+// ExtractVersion extracts a major.minor.patch version number from a typical CLI version flag output.
+//
+// minor and patch version numbers are both optional, treated as 0 if not found.
+func ExtractVersion(cliOutput string) (semver.Version, error) {
+	majorMinorPatch := regexp.MustCompile(`\d+\.\d+\.\d+`).FindString(cliOutput)
+	ver, err := semver.Parse(majorMinorPatch)
+	if err == nil {
+		return ver, nil
+	}
+
+	majorMinor := regexp.MustCompile(`(\d+)\.(\d+)`).FindStringSubmatch(cliOutput)
+	if len(majorMinor) >= 3 {
+		return semver.Version{
+			Major: parseUint(majorMinor[1]),
+			Minor: parseUint(majorMinor[2]),
+		}, nil
+	}
+
+	major := regexp.MustCompile(`\d+`).FindString(cliOutput)
+	if major != "" {
+		return semver.Version{Major: parseUint(major)}, nil
+	}
+
+	return semver.Version{}, fmt.Errorf("no valid version number found in %s", cliOutput)
+}
+
+func parseUint(s string) uint64 {
+	res, err := strconv.ParseUint(s, 10, 64)
 	if err != nil {
-		return semver, err
+		panic(err)
```

Comment:
Nit. Using log.panic() allow us to set a message with the error

Reply:
Will keep that in mind in the future. Think this is fine as-is given that it shouldn't be possible to get into this panic, and that the integer conversion error should have enough context.